internal/resources/oncall/resource_user_notification_rule.go

```diff
@@ -0,0 +1,328 @@
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var (
+	resourceUserNotificationRuleName = "grafana_oncall_user_notification_rule"
+
+	userNotificationRuleTypeOptions = []string{
+		"wait",
+		"notify_by_slack",
+		"notify_by_sms",
+		"notify_by_phone_call",
+		"notify_by_telegram",
+		"notify_by_email",
+		"notify_by_mobile_app",
+		"notify_by_mobile_app_critical",
+	}
+	userNotificationRuleTypeOptionsVerbal = strings.Join(userNotificationRuleTypeOptions, ", ")
+
+	// https://github.com/grafana/oncall/blob/6e0bebaa110a802187254321bd9c3c138fc590b6/engine/apps/base/models/user_notification_policy.py#L123-L129
+	userNotificationRuleDurationOptions = []int64{
+		60,      // 1 minute
+		60 * 5,  // 5 minutes
+		60 * 15, // 15 minutes
+		60 * 30, // 30 minutes
+		60 * 60, // 1 hour
+	}
+	userNotificationRuleDurationOptionsVerbal = strings.Trim(strings.Join(strings.Fields(fmt.Sprint(userNotificationRuleDurationOptions)), ", "), "[]")
+
+	// Check interface
+	_ resource.ResourceWithImportState = (*userNotificationRuleResource)(nil)
+)
+
+func resourceUserNotificationRule() *common.Resource {
+	return common.NewResource(
+		common.CategoryOnCall,
+		resourceUserNotificationRuleName,
+		resourceID,
+		&userNotificationRuleResource{},
+	).WithLister(oncallListerFunction(listUserNotificationRules))
+}
+
+type resourceUserNotificationRuleModel struct {
+	ID        types.String `tfsdk:"id"`
+	UserId    types.String `tfsdk:"user_id"`
+	Position  types.Int64  `tfsdk:"position"`
+	Duration  types.Int64  `tfsdk:"duration"`
+	Important types.Bool   `tfsdk:"important"`
+	Type      types.String `tfsdk:"type"`
+}
+
+type userNotificationRuleResource struct {
+	basePluginFrameworkResource
+}
+
+func (r *userNotificationRuleResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = resourceUserNotificationRuleName
+}
+
+func (r *userNotificationRuleResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"user_id": schema.StringAttribute{
+				MarkdownDescription: "User ID",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"position": schema.Int64Attribute{
+				MarkdownDescription: "Personal notification rules execute one after another starting from position=0. Position=-1 will put the escalation policy to the end of the list. A new escalation policy created with a position of an existing escalation policy will move the old one (and all following) down on the list.",
```

Comment:
`Position=-1 will put the escalation policy to the end of the list` this [doesn't work](https://github.com/grafana/oncall/blob/038d62788f66c5123174bd0efe41fa0ecd97c671/engine/common/ordered_model/serializer.py#L47) for Terraform (if there are multiple notification rules to be created, Terraform will issue multiple concurrent requests and there's no way to guarantee that the `-1` rule will actually be the last one).

Reply:
Good catch! Addressed in https://github.com/grafana/terraform-provider-grafana/pull/1653/commits/bdc35ec84cfbb5be2ca268da7996038996c1df38